lib/ui/discover/discover_detail_screen.dart

```diff
@@ -24,32 +28,137 @@ class _ArticleDetailState extends State<ArticleDetailScreen> {
     return Scaffold(
       body: StoreConnector<AppState, DiscoverViewModel>(
           builder: (_, viewModel) =>
-              _buildBody(context, viewModel.selectedArticle),
+              _buildList(context, viewModel.selectedArticleWithRelated, viewModel.getRelatedArticles),
           converter: (store) => DiscoverViewModel.fromStore(store)),
     );
   }
-
-  Widget _buildBody(BuildContext context, ArticleEntity selectedArticleData) {
+  Widget _buildList(BuildContext context, ArticleEntity selectedArticle, getRelatedArticles) {
     return Scaffold(
-          appBar: CustomAppBar.buildForDetail(selectedArticleData.title),
-          body : Container(
-
-              child: ListView(children: <Widget>[
+        appBar: CustomAppBar.buildForDetail(selectedArticle.title),
+        body: SingleChildScrollView(
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: <Widget>[
                 FadeInImage.assetNetwork(
-                    image: selectedArticleData.imageUrl,
+                    image: selectedArticle.imageUrl,
                     height: listImageHeight,
                     width: listImageWidth,
                     placeholder: Assets.IMAGE_PLACE_HOLDER,
                     fit: BoxFit.cover),
                 Padding(
-                    padding: Paddings.boxBigPadding(),
-                    child: Column(
-                      crossAxisAlignment: CrossAxisAlignment.start,
-                      children: <Widget>[
-                        Html(data: selectedArticleData.content),
-                      ],
-                    )),
-              ]))
-      );
+                    padding: Paddings.boxSmallPadding(),
+                    child: Text(selectedArticle.title,
+                        style: Styles.detailTitleTextStyle())),
+                Padding(
+                    padding: Paddings.boxSmallPadding(),
+                    child: Html(data: selectedArticle.content)),
+                Padding(
+                    padding: Paddings.boxSmallPadding(),
+                    child: Text("Related Articles", style: Styles.titleTextStyle()),
+                ),
+                buildRelated(context, selectedArticle.relatedArticles, getRelatedArticles),
+              ],
+            )));
   }
 }
+
+
+Widget buildRelated(BuildContext context, List<ArticleEntity> articlesList, getRelatedArticles) {
```

Comment:
Right now some widgets are duplicated because they are going to be deleted or rebuild with the redesing of the app, so when we redesign the app, we externalise the reusable widgets to make it more modular